src/Autofac/Features/Scanning/AssemblyExtensions.cs

```diff
@@ -0,0 +1,53 @@
+// Copyright (c) Autofac Project. All rights reserved.
```

Comment:
This all came out of the OpenGenericScanningRegistrationExtensions and ScanningRegistrationExtensions classes.